format/FormatNexusEigerDLS16M.py

```diff
@@ -51,6 +61,28 @@ def __init__(self, image_file, **kwargs):
         super(FormatNexusEigerDLS16M, self).__init__(image_file, **kwargs)
         self._dynamic_shadowing = self.has_dynamic_shadowing(**kwargs)
 
+    def get_detector(self, index=None):
+        if "_master.h5" not in self._image_file:
+            return self._detector()
+
+        # workaround for https://jira.diamond.ac.uk/browse/I03-365
+        # read the count limit from the meta file - if anything goes
+        # wrong, do nothing
+
+        detector = self._detector()
+
+        try:
+            meta = self._image_file.replace("_master.h5", "_meta.h5")
+            limit = get_count_limit_from_meta(meta)
```

Comment:
should we check for nonsensical values? `if limit > 0: for ...`?

Reply:
Yes, probably

Reply:
Added assert that limit > 0